flamedisx/x1t_sr1.py

```diff
@@ -15,21 +19,126 @@
 ##
 # Yield maps
 ##
-
-
 s1_map, s2_map = [
     fd.InterpolatingMap(fd.get_resource(fd.pax_file(x)))
     for x in ('XENON1T_s1_xyz_ly_kr83m-SR1_pax-664_fdc-adcorrtpf.json',
               'XENON1T_s2_xy_ly_SR1_v2.2.json')]
 
+##
+# Parameters
+##
+def_g1 = 0.142
+def_g2 = 11.4 # don't divide by weird shits. will explode downstream.
+
+def_frac_top = 0.63 # fraction of light from top array
+def_p_dpe = 0.219
+def_extract_eff = 0.96
+
+def_elife = 641e3
+def_drift_vel = 1.34 * 1e-4   # cm/ns, from analysis paper II
+
+def_field = 81.
+
+def_recon_pivot = 0.49
 
 ##
-# Flamedisx sources
+# Loading Pax reconstruction bias
+##
+dummy_base = os.path.dirname(os.path.realpath(__file__)) + '/dummy_maps/'
+
+path_recon_bias_mean_s1 = ['/home/peaelle42/software/bbf/bbf/data/ReconstructionS1BiasMeanLowers_SR1_v2.json',
+         '/home/peaelle42/software/bbf/bbf/data/ReconstructionS1BiasMeanUppers_SR1_v2.json']
+path_recon_bias_mean_s2 = ['/home/peaelle42/software/bbf/bbf/data/ReconstructionS2BiasMeanLowers_SR1_v2.json',
+         '/home/peaelle42/software/bbf/bbf/data/ReconstructionS2BiasMeanUppers_SR1_v2.json']
+
+path_dummy_zeros_s1 = ['dummy_zeros_s1.json', 'dummy_zeros_s1.json']
+path_dummy_zeros_s1 = [dummy_base+x for x in path_dummy_zeros_s1]
+path_dummy_zeros_s2 = ['dummy_zeros_s2.json', 'dummy_zeros_s2.json']
+path_dummy_zeros_s2 = [dummy_base+x for x in path_dummy_zeros_s2]
+
+def read_bias_tf(path_bag):
+    # Achtung: 
+    # Fundamentally assumes upper and lower bounds have exactly the same support definition
+    data_bag = []
+    yy_ref_bag = []
+    for loc_path in path_bag:
+        with open(loc_path) as json_file:
+            tmp = json.load(json_file)
+            yy_ref_bag.append(tf.convert_to_tensor(tmp['map'], dtype=fd.float_type()))
+            data_bag.append(tmp)
```

Comment:
These two lines could be deindented so the json can be closed immediately after loading tmp. (and, more importantly, it's clear you're done with the file after the first line)

Reply:
Edited.